packages/orchestration/src/exos/local-orchestration-account.js

```diff
@@ -265,32 +263,48 @@ export const prepareLocalOrchestrationAccountKit = (
        */
       returnVoidWatcher: {
         /**
-         * @param {Record<unknown, unknown>[]} results
+         * @param {any} _result
          */
-        onFulfilled(results) {
-          results.length === 1 ||
-            Fail`expected exactly one result; got ${results}`;
-          trace('Result', results[0]);
+        onFulfilled(_result) {
           return undefined;
         },
       },
+      /**
+       * handles a request for balance from a bank purse and returns the balance
+       * as a Chain Amount
+       */
+      getBalanceWatcher: {
+        /**
+         * @param {Amount<'nat'>} natAmount
+         * @param {{ denom: DenomAmount['denom'] }} ctx
+         * @returns {DenomAmount}
+         */
+        onFulfilled(natAmount, { denom }) {
+          return harden({ denom, value: natAmount.value });
+        },
+      },
       holder: {
-        /** @type {OrchestrationAccount<any>['getBalance']} */
-        async getBalance(denomArg) {
+        /**
+         * TODO: balance lookups for non-vbank assets
+         *
+         * @type {PromiseToVow<OrchestrationAccount<any>['getBalance']>}
+         */
+        getBalance(denomArg) {
           // FIXME look up real values
           // UNTIL https://github.com/Agoric/agoric-sdk/issues/9211
           const [brand, denom] =
             typeof denomArg === 'string'
               ? [/** @type {any} */ (null), denomArg]
               : [denomArg, 'FIXME'];
 
-          const natAmount = await E.when(
+          return watch(
             E(this.state.account).getBalance(brand),
+            this.facets.getBalanceWatcher,
+            { denom },
```

Comment:
any particular reason why you pass `denom` in an embedded record rather than as a sole argument? Consistency?  I check reflexively because it is more allocations :)

Reply:
but perhaps clearer

Reply:
Context is a single argument, so I’ve been treating it like an opts bag for consistency/readability.

This is good feedback, though. These changes worth considering as we get closer to completion cc @turadg

Reply:
> ... These changes worth considering as we get closer to completion ...

I hope any plans to re-consider code we've written are represented by open issues.

Reply:
Not worth a ticket so updated here: [69a6e21](https://github.com/Agoric/agoric-sdk/pull/9591/commits/69a6e2157b8d1223adfd73ff7d8927ea67628e3b)